odoo/addons/base/ir/ir_mail_server.py

```diff
@@ -138,6 +139,17 @@ class IrMailServer(models.Model):
                           "specified for outgoing emails (To/Cc/Bcc)")
 
     name = fields.Char(string='Description', required=True, index=True)
+    smtp_from = fields.Char(
+        string='Email From',
+        help='Set this in order to email from a specific address when using '
+             'this SMTP server.',
+    )
+    smtp_from_only_dmarc = fields.Boolean(
```

Comment:
Note that the introduction of this feature also introduced a new Python dependency (`gs.dmarc`), which may be undesirable in versions 10 & 11 because they have already been released. 

I can break this part off for master if the consensus agrees with above.

Reply:
Removed and moved to #20915